docs/ConcurrencyRoadmap.md

```diff
@@ -0,0 +1,265 @@
+
+So the patterns shown here are good, but expressing them in Swift loses important structure and creates problems. The solution is to bring those patterns into the language. That will reduce boilerplate and let the language make the patterns safe, eliminating bugs and giving programmers the confidence to use concurrency more pervasively. It will also give us a chance to improve the performance of concurrent code.
+
+Here is the above code rewritten using our proposed new syntax:
+
+```swift
+internal func refreshPlayers() async {
+  players = await gameSession.allPlayers().map(\.nickname)
+}
+```
+
+Things to note about this example:
+
+- `refreshPlayers` is now an `async` function.
+- `allPlayers` is also an `async` function, and _returns_ its result instead of passing it into a completion handler.
+- Because of this, we can use expression composition to call the `map` function directly on the returned value.
+- The `await` keyword appears before the expression calling `allPlayers`, indicating that the `refreshPlayers` function can become _suspended_ at this point.
+- `await` works similarly to `try`, in that it need only appear once at the start of an expression that can suspend, not directly in front of every call that can suspend within that expression.
+- The explicit `self.` has been eliminated from property accesses, because no escaping closures capturing `self` are required.
+- The accesses to the properties `allPlayers` and `players` now cannot have data races.
+
+To understand how the the last point is achieved we must step out a layer and look at how queues are used to protect state.
+
+The original code was a method on a class that used `refreshQueue` to protect its internal state:
+
+```swift
+class PlayerRefreshController {
+  var players: [String] = []
+  var gameSession: GameSession
+  var refreshQueue = DispatchQueue(label: "PlayerRefresh")
+    
+  func refreshPlayers(completion: (() -> Void)? = nil) { 
+    ... 
+  }
+}
+```
+
+This is a common pattern: a class with a private queue and some properties that should only be accessed on the queue. We replace this manual queue management with an actor class:
+
+```swift
+actor class PlayerRefreshController {
+  var players: [String] = []
+  var gameSession: GameSession
+
+  func refreshPlayers() async { ... }
+}
+```
+
+Things to note about this example:
+
+- Declaring a class to be an actor is similar to giving a class a private queue and synchronizing all access to its private state through that queue.
+- Because this synchronization is now understood by the compiler, you cannot _forget_ to use the queue to protect state: the compiler will ensure that you are running on the queue in the class's methods, and it will prevent you from accessing the state outside those methods.
```

Comment:
```suggestion
- Because this synchronization is now understood by the compiler, you _won’t forget_ to use the queue to protect state: the compiler will ensure that you are running on the queue in the class's methods, and it will prevent you from accessing the state outside those methods.
```